sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala

```diff
@@ -172,15 +173,23 @@ case class GpuJsonToStructs(
         // Step 1: verify and preprocess the data to clean it up and normalize a few things
         // Step 2: Concat the data into a single buffer
         val (isNullOrEmpty, combined) = cleanAndConcat(input.getBase)
+
         withResource(isNullOrEmpty) { isNullOrEmpty =>
           // Step 3: setup a datasource
           val table = withResource(new JsonDeviceDataSource(combined)) { ds =>
             // Step 4: Have cudf parse the JSON data
-            cudf.Table.readJSON(cudfSchema, jsonOptions, ds)
+            try {
+              cudf.Table.readJSON(cudfSchema, jsonOptions, ds)
+            }catch{
+              case e @ (_: IllegalStateException | _: CudfException) =>
+                logError("Rapids currently does not support all JSON to struct cases of " +
+                  "from_json. Consider turning it off by setting " +
+                  "spark.rapids.sql.expression.JsonToStructs=false")
+                throw new RuntimeException("Failed to parse JSON data using cuDF", e)
```

Comment:
nit: I'm not sure RuntimeException is the ideal one to use here, but I'm not sure there is a better one.